cmake/multi_image.cmake

```diff
@@ -150,6 +150,10 @@ function(zephyr_add_external_image_from_source name sourcedir)
 
       # Add dependency for the benefit of make.
       add_dependencies(zephyr_interface ${spm_veneers_lib})
+
+      # TODO: Find out if the below line makes the above line
+      # redundant.
+      add_dependencies(zephyr_interface SPM_subimage)
```

Comment:
Please, do not at the dependency to the external project.
There is no need to rerun everything just because External project is rerun, only if the output from external project changes, in this case: `${spm_veneers_lib}` the dependents should be recompiled / relinked.

Reply:
depending on the byproducts also allows for creation of the veneers lib in other ways, as example, a pre-built lib.

Reply:
let's discuss this in person